editors/vscode/src/browser-lsp-worker.ts

```diff
@@ -1,46 +1,34 @@
 // Copyright © SixtyFPS GmbH <info@slint.dev>
 // SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
 
+import slint_init, * as slint_lsp from "../out/slint_lsp_wasm.js";
+import slint_wasm_data from "../out/slint_lsp_wasm_bg.wasm";
+import { InitializeParams, InitializeResult } from "vscode-languageserver";
 import {
     createConnection,
     BrowserMessageReader,
     BrowserMessageWriter,
 } from "vscode-languageserver/browser";
 
-import { InitializeParams, InitializeResult } from "vscode-languageserver";
-
-import slint_init, * as slint_lsp from "../out/slint_lsp_wasm.js";
-import slint_wasm_data from "../out/slint_lsp_wasm_bg.wasm";
-
 slint_init(slint_wasm_data).then((_) => {
-    const messageReader = new BrowserMessageReader(self);
-    const messageWriter = new BrowserMessageWriter(self);
-    const connection = createConnection(messageReader, messageWriter);
+    const reader = new BrowserMessageReader(self);
+    const writer = new BrowserMessageWriter(self);
 
     let the_lsp: slint_lsp.SlintServer;
 
+    const connection = createConnection(reader, writer);
+
     function send_notification(method: string, params: any): boolean {
         connection.sendNotification(method, params);
         return true;
     }
 
-    async function load_file(path: string): Promise<string> {
-        const contents: Uint8Array = await connection.sendRequest(
-            "slint/load_file",
-            path,
-        );
-        return new TextDecoder().decode(contents);
```

Comment:
Are you sure about removing the decoding part? AFAICS the counter-part of this, the request handler's use of `vscode.workspace.fs.readFile()`, still returns a UInt8Array that needs decoding into the JS utf-16 string.

Reply:
Looking at your branch, I guess you could continue with returning a UInt8Array here if you changed the load_file signature, implementation, and generated `ImportCallbackFunction` signature to expect that, and then do the utf-8 decoding in Rust. But I think you need to either revert this part or change the wasm side that currently expects a JS string.